### ISSUE TITLE
Queue mode documentation

### ISSUE THREAD

**Issue by super-ienien** [SuperFlyTV/casparcg-connection#45]
Hi,

I would love to see a doc for the Queue Mode.

In my current project i'm using the as3 server-connection (with some homemade improvements) to communicate with casparCG but i will rebuild th whole thing in javascript using casparcg-connection. I need  to send a lot of commands without waiting the response from caspar. I think what i need is the SALVO mode right ? also what is the SMART mode ? it is present in the ENUM but i don't see it used in the code.

**Comment by jesperstarkar** [comment]
For now only SEQUENTIAL is properly implemented. I have removed SALVO and SMART for now, but will share my thoughts with you here:

- The main problem is that the AMCP socket server is asynchronous, and we have no way of identifying the response to match against the request.
- Going with a SALVO mode now, means that we need to let go of all response statuses. We will en up with a list of sent commands, and a list of received responses, with no relationships between them. And that would probably be just fine for some use cases, your's included. As I said, this is currently disabled, and I will add this back once I figure out how it should work. SALVO would then work after the pricipal: "sent all commands as fast as possible, and just keep the responses for logging and processing state-data (config, version, paths, cls, tls etc.). I think we will resolve all Commands to success as we send them, then just firing standalone response objects as events or callbacks, perhaps keeping a short buffered list over last x responses in memory.
- A "SMART" mode, not startet yet at all, might not be started ever. The idea was to filter on the signature of the responses of each kind of command, sending commands in parallel as long as they were distinguishable based on the respons. So a PLAY and a MIXER could go in parallel, but two PLAYs couldn't. I just realised that the problem here is that the ERROR responses are similar between different kinds of commands, so any ERROR would mess up the separation.
- A last method is to open multiple sockets in parallel over different ports, load-balancing the commands. I think they still will be nested together into a single command queue internally in CasparCG Server, but we might (I dont't know, haven't tested this) be able to have multiple sockets in parallel and that way handle multiple SEQUENTIAL queues. 

Regarding documentation: you're right, time to start documenting core features that start to mature.
